connectors/sources/sandfly.py

```diff
@@ -0,0 +1,828 @@
+                raise
+            elif exception.status == 404:
+                msg = "Resource Not Found"
+                raise ResourceNotFound(msg) from exception
+            else:
+                raise
+        except Exception:
+            raise
+
+    @asynccontextmanager
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+        skipped_exceptions=ResourceNotFound,
+    )
+    async def _post(self, absolute_url, payload):
+        try:
+            access_token = await self._my_token.get()
+            headers = {
+                "Accept": "application/json",
+                "Content-type": "application/json",
+                "Authorization": f"Bearer {access_token}",
+            }
+
+            async with self._http_session.post(
+                url=absolute_url, headers=headers, data=json.dumps(payload)
+            ) as response:
+                yield response
+        except ClientResponseError as exception:
+            if exception.status == 401:
+                await self._my_token.get(is_cache=False)
+                raise
+            elif exception.status == 404:
+                msg = "Resource Not Found"
+                raise ResourceNotFound(msg) from exception
+            else:
+                raise
+        except Exception:
+            raise
+
+    async def content_get(self, url):
+        try:
+            async with self._get(absolute_url=url) as response:
+                return await response.text()
+        except Exception as exception:
+            self._logger.warning(
+                f"Content for {url} is being skipped. Error: {exception}."
+            )
+
+    async def content_post(self, url, payload):
+        try:
+            async with self._post(absolute_url=url, payload=payload) as response:
+                return await response.text()
+        except Exception as exception:
+            self._logger.warning(
+                f"Content for {url} is being skipped. Error: {exception}."
+            )
+
+
+class SandflyClient:
+    def __init__(self, configuration):
+        self._sleeps = CancellableSleeps()
+        self._logger = logger
+
+        self.configuration = configuration
+
+        self.server_url = self.configuration["server_url"]
+        self.verify_ssl = self.configuration["verify_ssl"]
+
+        t_connector = aiohttp.TCPConnector(
+            family=socket.AF_INET, verify_ssl=self.verify_ssl
+        )
+        self.http_session = aiohttp.ClientSession(
+            connector=t_connector, raise_for_status=True
+        )
+
+        self.my_token = SandflyAccessToken(
+            http_session=self.http_session,
+            configuration=configuration,
+            logger_=self._logger,
+        )
+        self.my_client = SandflySession(
+            http_session=self.http_session,
+            token=self.my_token,
+            logger_=self._logger,
```

Comment:
Why prefix with `my_*`? Is there a need to also access these variables from outside of the class?